libp2p/switch.nim

```diff
@@ -577,11 +617,27 @@ proc newSwitch*(peerInfo: PeerInfo,
       await conn.close()
     trace "Muxed connection done", conn
 
-  result.mount(identity)
+  switch.mount(identity)
   for key, val in muxers:
-    val.streamHandler = result.streamHandler
+    val.streamHandler = switch.streamHandler
     val.muxerHandler = proc(muxer: Muxer): Future[void] =
-      s.muxerHandler(muxer)
+      switch.muxerHandler(muxer)
+
+  proc peerEventHandler(peerId: PeerID, event: ConnEvent) {.async, gcsafe.} =
+    ## peer event handler
+    ##
+
+    if peerId notin switch.connManager:
```

Comment:
technically, can get the `connCount` here to avoid the additional lookups in the table (once for `in` and once for `count`)

Reply:
not sure what you mean here?